src/FormBuilder/resources/js/form-builder/src/blocks/fields/phone/Edit.tsx

```diff
@@ -0,0 +1,66 @@
+import classnames from 'classnames';
+import {BlockEditProps} from '@wordpress/blocks';
+import {BaseControl, PanelBody, PanelRow} from '@wordpress/components';
+import {useEffect} from 'react';
+import {getFormBuilderWindowData} from '@givewp/form-builder/common/getWindowData';
+import Label from '@givewp/form-builder/blocks/fields/settings/Label';
+import Required from '@givewp/form-builder/blocks/fields/settings/Required';
+import {__} from '@wordpress/i18n';
+import {InspectorControls} from '@wordpress/block-editor';
+
+/**
+ * @unreleased
+ */
+export default function Edit({attributes: {label, required}, setAttributes}: BlockEditProps<any>) {
+    const intlTelInputId: string = 'give-form-builder-phone-input';
+
+    useEffect(() => {
+        const {intlTelInputSettings} = getFormBuilderWindowData();
+
+        const css = document.createElement('link');
+        css.href = intlTelInputSettings.cssUrl;
+        css.rel = 'stylesheet';
+        document.body.appendChild(css);
+
+        const script = document.createElement('script');
+        script.src = intlTelInputSettings.scriptUrl;
+        script.async = true;
+        script.onload = () => {
+            // @ts-ignore
+            window.intlTelInput(document.querySelector('#' + intlTelInputId), {
+                showSelectedDialCode: intlTelInputSettings.showSelectedDialCode,
+                strictMode: intlTelInputSettings.strictMode,
+                utilsScript: intlTelInputSettings.utilsScriptUrl,
+                initialCountry: intlTelInputSettings.initialCountry,
+                i18n: intlTelInputSettings.i18n,
+            });
+        };
+        document.body.appendChild(script);
+
+        return () => {
+            document.body.removeChild(css);
+            document.body.removeChild(script);
+        };
+    }, []);
+
+    return (
+        <>
+            <div className={classnames({'give-is-required': required})}>
+                <BaseControl id={'give-form-builder-phone-label'} label={label}>
+                    <input id={intlTelInputId} type="text" />
+                </BaseControl>
+            </div>
+
+            <InspectorControls>
+                <PanelBody title={__('Field Settings', 'give')} initialOpen={true}>
+                    <PanelRow>
+                        <Label label={label} setAttributes={setAttributes} />
+                    </PanelRow>
+                    <PanelRow>
+                        <Required isRequired={required} setAttributes={setAttributes} />
```

Comment:
This required toggle was not working for me:

<img width="327" alt="Screenshot 2024-04-11 at 1 04 00 PM" src="https://github.com/impress-org/givewp/assets/10138447/a419b782-554b-4133-95d4-ac88f95762e2">